### ISSUE TITLE
X axis values larger than int.Max do not work for Cartesian charts

### ISSUE THREAD

**Issue by buybackoff** [beto-rodriguez/Live-Charts#534]
Cannot make large X values work, but every time I change the values to fit `Int32` range it works without other changes.  I believe this is caused by rounding `double`s using int cast instead of `Math.Floor/Ceiling`, e.g. [here](https://github.com/beto-rodriguez/Live-Charts/blob/master/Core40/Charts/ChartCore.cs#L678). The workaround is to set the `InitialDateTime` for an axis in such a way that all values are within the Int32 range, but this is inconvenient and the entire `double` range should be supported. E.g. total number of seconds for `DateTime` is larger than Int32.MaxValue since the year 0068. Or if this is by design, this should be documented in bold :)

**Comment by kpnz** [comment]
This fix would be good. I'm currently working around by converting ticks smaller than int32 too.

My code was throwing this exception in:
`CartesianChartCore.PrepareAxes()`
```
                if (Math.Abs(xi.BotLimit - xi.TopLimit) < xi.S * .01)
                {
                    if (Math.Abs(xi.PreviousBot - xi.PreviousTop) < xi.S * .01)
                    {
                        if (double.IsNaN(xi.MinValue)) xi.BotLimit -= xi.S;
                        else xi.BotLimit = xi.MinValue;

                        if (double.IsNaN(xi.MaxValue)) xi.TopLimit += xi.S;
                        else xi.TopLimit = xi.MaxValue;

                        if (Math.Abs(xi.BotLimit - xi.TopLimit) < xi.S * .01 && !View.IsInDesignMode)
                            throw new LiveChartsException("One axis has an invalid range, it is or it is " +
                                                          "tends to zero, please ensure your axis has a valid " +
                                                          "range");
```
when there was only one point that is larger than int32 the twiddle either side had no effect..

**Comment by beto-rodriguez** [comment]
Fixed with #534 thanks
